Modules/gcmodule.c

```diff
@@ -31,6 +31,56 @@
 #include "pydtrace.h"
 #include "pytime.h"             /* for _PyTime_GetMonotonicClock() */
 
+#define GC_DEBUG (0)  /* More asserts */
+
+// Bit 0 of gc_prev is used for _PyGC_PREV_MASK_FINALIZED in objimpl.h
+// These two mask bits are used only between GC.
+#define MASK_COLLECTING              (1 << 1)
+#define MASK_TENTATIVELY_UNREACHABLE (1 << 2)
+
+#define IS_TENTATIVELY_UNREACHABLE(o) ( \
+    (_Py_AS_GC(o)->gc.gc_prev & MASK_TENTATIVELY_UNREACHABLE) != 0)
+
+static inline int
+gc_is_collecting(PyGC_Head *g)
+{
+    return (g->gc.gc_prev & MASK_COLLECTING) != 0;
+}
+
+static inline void
+gc_clear_masks(PyGC_Head *g)
```

Comment:
You don't clear a mask but flags, no? Maybe: gc_clear_internal_flags()?

Maybe rename "internal flags" to "collection flags"? I don't know.

Reply:
Since I removed `MASK_TENTATIVELY_UNREACHABLE` from gc_prev, I renamed this to `gc_clear_collecting`.